analysers/analyser_osmosis_polygon_intersects.py

```diff
@@ -0,0 +1,316 @@
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+CREATE TEMP TABLE mobility_ways AS
+SELECT
+  *
+FROM (
+  SELECT
+    id,
+    linestring,
+    'highway' AS type,
+    tags
+  FROM
+    highways
+  WHERE
+    level <= 3
+  UNION ALL
+  SELECT
+    id,
+    linestring,
+    CASE
+      WHEN tags?'waterway' THEN 'waterway'
+      WHEN tags?'railway' THEN 'railway'
+      WHEN tags?'aeroway' THEN 'aeroway'
+    END AS type,
+    tags
+  FROM
+    ways
+  WHERE
+    tags != ''::hstore AND
+    (
+      (
+        tags?'railway' AND
+        tags->'railway' NOT IN ('abandoned', 'construction', 'disused', 'dismantled', 'miniature', 'proposed', 'razed', 'subway') AND
+        NOT is_polygon
+      ) OR (
+        tags?'waterway' AND
+        tags->'waterway' IN ('river', 'canal') AND -- big waterways
+        (NOT tags?'seasonal' OR tags->'seasonal' = 'no') AND
+        (NOT tags?'intermittent' OR tags->'intermittent' = 'no')
+      ) OR (
+        tags?'aeroway' AND
+        tags->'aeroway' IN ('apron', 'helipad', 'runway', 'taxilane', 'taxiway') -- aircraft moving places
+      )
+    )
+  ) AS t
+WHERE
+  NOT tags?'abandoned' AND
+  NOT tags?'disused' AND
+  NOT tags?'razed' AND
+  (NOT tags?'bridge' OR tags->'bridge' = 'no') AND
+  (NOT tags?'tunnel' OR tags->'tunnel' = 'no')
+"""
+
+sql11 = """
+CREATE TEMP TABLE landusage AS
+SELECT
+  id,
+  linestring,
+  ST_MakePolygon(linestring) AS poly,
+  CASE
+    WHEN tags?'landuse' THEN 'landuse'
+    WHEN tags?'natural' THEN 'natural'
+  END AS landusekey,
+  CASE
+    WHEN tags?'landuse' THEN tags->'landuse'
+    WHEN tags?'natural' THEN tags->'natural'
+  END AS landusevalue,
+  tags,
+  FALSE AS is_multipolygon
+FROM
+  ways
+WHERE
+  is_polygon AND
+  tags != ''::hstore AND
+  (tags?'natural' OR tags?'landuse') AND
+  -- Leave minipolygons for analyser osmosis_polygon_small
+  ST_Area(ST_MakePolygon(ST_Transform(linestring, {proj}))) >= 20
+UNION ALL
+SELECT
+  ways.id,
+  ways.linestring,
+  -- Better would be to parse inner and outer segments and create an SQL multipolygon
+  -- For now, just use a 0.0000000001m buffer zone around the linestring
+  ST_Buffer(ways.linestring, 0.0000000001, 'quad_segs=1 endcap=flat join=bevel') AS poly,
```

Comment:
That in degrees, not in meters.

Reply:
I assume you're referring to the 0.0000000001m? If I read the docs correctly, it should be the radius, not the degrees, and thus be a distance unit, not a degrees value?

> For geometry, the distance is specified in the units of the Spatial Reference System of the geometry. For geography, the distance is specified in meters.

https://postgis.net/docs/ST_Buffer.html
(Although my assumption of meter is also wrong as I'm not passing it through ST_Transform; it just had to be very small)

Reply:
I've removed the bad comment about meters :)